File: system-test/dotnet/Dsl.Core/UseCase/External/Clock/UseCases/GoToClock.cs
using Dsl.Core.External.Clock.UseCases.Base;
using Driver.Port.External.Clock;
using Driver.Port.External.Clock.Dtos;
using Common;
using Dsl.Core.Shared;

namespace Dsl.Core.External.Clock.UseCases;

public class GoToClock : BaseClockCommand<VoidValue, VoidVerification>
{
    public GoToClock(IClockDriver driver, UseCaseContext context)
        : base(driver, context)
    {
    }

    public override async Task<ClockUseCaseResult<VoidValue, VoidVerification>> Execute()
    {
        var result = await _driver.GoToClockAsync();

        return new ClockUseCaseResult<VoidValue, VoidVerification>(
            result,
            _context,
            (response, ctx) => new VoidVerification(response, ctx));
    }
}

File: system-test/java/src/main/java/com/optivem/shop/dsl/core/usecase/external/clock/usecases/GoToClock.java
package com.optivem.shop.dsl.core.usecase.external.clock.usecases;

import com.optivem.shop.dsl.driver.port.external.clock.ClockDriver;
import com.optivem.shop.dsl.core.usecase.external.clock.usecases.base.BaseClockUseCase;
import com.optivem.shop.dsl.core.shared.UseCaseResult;
import com.optivem.shop.dsl.core.shared.UseCaseContext;
import com.optivem.shop.dsl.core.shared.VoidVerification;

public class GoToClock extends BaseClockUseCase<Void, VoidVerification> {
    public GoToClock(ClockDriver clockDriver, UseCaseContext useCaseContext) {
        super(clockDriver, useCaseContext);
    }

    @Override
    public UseCaseResult<Void, VoidVerification> execute() {
        var result = driver.goToClock();
        return new UseCaseResult<>(result, context, VoidVerification::new);
    }
}
